### PR TITLE
Deprecate `is_bsr_head` field

### PR DESCRIPTION
**Pull request by unmultimedio**

### CODE REVIEW

**Review by doriable — APPROVED** [review]

**Review by saquibmian — APPROVED** [review]

**Review by saquibmian — APPROVED** [review]

File: proto/buf/alpha/registry/v1alpha1/resolve.proto [hunk 1]
```diff
@@ -120,7 +120,7 @@ message LocalModuleResolveResult {
   ResolvedReferenceType resolved_reference_type = 3;
   // Indicates whether this reference is resolvable through BSR_HEAD.
   // False indicates a non-default branch (formerly draft) commit.
-  bool is_bsr_head = 4;
+  bool is_bsr_head = 4 [deprecated = true];
```
Comment (bufdev) [inline proto/buf/alpha/registry/v1alpha1/resolve.proto:123]: Are we still using this field?

Comment (unmultimedio) [reply proto/buf/alpha/registry/v1alpha1/resolve.proto:123]: We have a PR open where we stop using it. The RPC still returns it though.

Comment (paul-sachs) [reply proto/buf/alpha/registry/v1alpha1/resolve.proto:123]: Nope, nothing in the UI is using this field.

Comment (bufdev) [reply proto/buf/alpha/registry/v1alpha1/resolve.proto:123]: Can we delete and reserve it, similar to other changes we've made?

Comment (unmultimedio) [reply proto/buf/alpha/registry/v1alpha1/resolve.proto]: Done

File: proto/buf/alpha/registry/v1alpha1/resolve.proto [hunk 2]
```diff
@@ -118,9 +118,8 @@ message LocalModuleResolveResult {
   buf.alpha.registry.v1alpha1.LocalModulePin pin = 2;
   // The type the reference resolved as.
   ResolvedReferenceType resolved_reference_type = 3;
-  // Indicates whether this reference is resolvable through BSR_HEAD.
-  // False indicates a non-default branch (formerly draft) commit.
-  bool is_bsr_head = 4;
+  reserved 4;
```
Comment (bufdev) [inline proto/buf/alpha/registry/v1alpha1/resolve.proto:121]: Every other declaration of `reserved` in this repository save one puts the `reserved` declarations at the top of the message, can you follow this convention?

**Review by bufdev — APPROVED** [review]
I'm approving this so there's no more back and forth, but please fix the comment up

**Pull request merged**